halo2-ecc/src/ecc/mod.rs

```diff
@@ -1014,6 +1036,36 @@ impl<'chip, F: BigPrimeField, FC: FieldChip<F>> EccChip<'chip, F, FC>
 where
     FC: Selectable<F, FC::FieldPoint>,
 {
+    pub fn sum_safe<C>(
+        &self,
+        ctx: &mut Context<F>,
+        points: impl IntoIterator<Item = EcPoint<F, FC::FieldPoint>>,
+    ) -> EcPoint<F, FC::FieldPoint>
+    where
+        C: CurveAffineExt<Base = FC::FieldType>,
+    {
+        let rand_point = self.load_random_point::<C>(ctx);
+        let rand_point2 = self.load_random_point::<C>(ctx);
+        let zero = ctx.load_constant(F::ZERO);
+        let rand_point = into_strict_point(self.field_chip, ctx, rand_point);
+        let neg_rand_point = self.negate_strict(ctx, rand_point.clone());
+        let mut acc = StrictEcPoint::from(neg_rand_point.clone());
+        for point in points {
+            let point_is_inf = self.is_infinity(ctx, point.clone());
+            let addend = self.select(ctx, rand_point2.clone(), point.clone(), point_is_inf);
+            let _acc = self.add_unequal(ctx, acc.clone(), addend.clone(), true);
+            let _acc = self.select(ctx, acc.clone().into(), _acc, point_is_inf);
+            acc = into_strict_point(self.field_chip, ctx, _acc);
```

Comment:
the `into_strict_point` can be skipped: the point of `ComparableEcPoint` is exactly to automate this process:
* `_acc` implements `Into<ComparableEcPoint>` and it will be the `NonStrict` variant
* `add_unequal` will see this and automatically call `enforce_less_than`